app/controllers/policy_classes_controller.rb

```diff
@@ -0,0 +1,75 @@
+class PolicyClassesController < PlanningApplicationsController
+  before_action :set_planning_application
+  before_action :set_policy_class, only: %i[show update destroy]
+
+  def part; end
+
+  def new
+    @part = params[:part]
+
+    redirect_to part_new_policy_assessment_path(@planning_application), alert: "Please choose one of the policy parts" if @part.blank?
+  end
+
+  def create
+    class_ids = policy_class_params[:policy_classes].reject(&:blank?)
+
+    classes = PolicyClass
+                .classes_for_part(params[:part])
+                .select { |c| class_ids.include? c.id }
+                .each(&:stamp_status!)
+
+    @planning_application.policy_classes += classes
+
+    if @planning_application.save
+      redirect_to @planning_application, notice: "Policy classes have been successfully added"
```

Comment:
we need an error here as well if the policy_classes are empty - currently when leaving all the boxes unselected and pressing "Add classes" the success message is displayed as we aren't forcing the user to choose any:

![image](https://user-images.githubusercontent.com/9452321/134502824-38737036-df7b-41b1-b184-db9bc7f445e3.png)

Reply:
Cheers, will do. Also, no need for screenshots every time :smile:

Reply:
sorry about the endless screenshots its a habit from doing tech documentation but I realise totally unnecessary for this 😆